From 3307dac840a95e6e443603d9c360a843aa314558 Mon Sep 17 00:00:00 2001
From: Yonghong Song <ysong@broadcom.com>
Date: Mon, 23 Apr 2012 13:45:15 -0700
Subject: [PATCH 563/762] Fix: Restrict the kernel init thread to node 0

  o This way, the multi-node without NUMA can boot properly

Based on Broadcom SDK 2.3.

Signed-off-by: Yonghong Song <ysong@broadcom.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 init/main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/init/main.c b/init/main.c
index 6c45c22..7d74ab1 100644
--- a/init/main.c
+++ b/init/main.c
@@ -879,7 +879,7 @@ static int __init kernel_init(void * unused)
 	smp_init();
 	sched_init_smp();
 
-#if defined(CONFIG_NLM_XLP) && defined(CONFIG_NUMA)
+#if defined(CONFIG_NLM_XLP)
 	/* On XLP Ax, PIC device specific registers cannot be accessed
 	 * cross different nodes. So the kernel_init has to run on node 0.
 	 * On XLP B0, the flash device cannot be accessed cross node, so flash
-- 
1.7.0.4

